src/modules/70-pipeline/components/PipelineSteps/Steps/JiraCreate/JiraCreateDeploymentMode.tsx

```diff
@@ -123,6 +130,31 @@ function FormContent(formContentProps: JiraCreateDeploymentModeFormContentInterf
       }
     }
   }, [projectMetaResponse?.data])
+
+  useEffect(() => {
+    if (issueKeyFixedValue && projectMetadata?.issuetypes) {
+      const issueTypeData = projectMetadata?.issuetypes[issueKeyFixedValue as string]
+      const fieldKeys = Object.keys(issueTypeData?.fields || {})
+      const jiraSelectedFields: JiraFieldNGWithValue[] = []
+      fieldKeys.forEach(keyy => {
+        const fieldObject = issueTypeData?.fields[keyy]
+        if (fieldObject && keyy !== 'Summary' && keyy !== 'Description') {
+          template?.spec?.fields?.filter(_field =>
+            _field.name === keyy
+              ? jiraSelectedFields.push({
+                  ...fieldObject,
+                  value: _field.value
+                })
+              : null
+          )
+          return jiraSelectedFields
+        }
+      })
+      // console.log(jiraSelectedFields)
+      setSelectedField(jiraSelectedFields)
+    }
+  }, [issueKeyFixedValue, projectMetadata])
```

Comment:
Codacy has a fix for the issue: [React Hook useEffect has a missing dependency: 'template?.spec?.fields'. Either include it or remove the dependency array.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232288)
```suggestion
  }, [issueKeyFixedValue, projectMetadata, template?.spec?.fields])
```